packages/grid/data-grid/src/DataGrid.tsx

```diff
@@ -80,7 +80,8 @@ DataGrid2.propTypes = {
       );
     }
   }),
-  disableColumnReorder: chainPropTypes(PropTypes.bool, (props) => {
+  disableColumnReorder: chainPropTypes(PropTypes.bool, (props: DataGridProps) => {
+    // @ts-expect-error not allowed
```

Comment:
why is there a ts error here and what's the error? Can we fix it?

Reply:
The error should be present. The prop-types raise a warning for a value TypeScript rejects. It could almost replace:

https://github.com/mui-org/material-ui-x/blob/1a3bf5f47d3871a0f3fd7f1ddaa7fd93f7b16c72/packages/grid/data-grid/src/DataGrid.spec.tsx#L10-L11